### ISSUE TITLE
running commands from a view does not affect entire repo

### ISSUE THREAD

**Issue by Enteleform** [titoBouzout/SideBarGit#40]
This issue is regarding the `side_bar_git_add_commit_push` command.

If run from the sidebar context menu after right-clicking on the repo directory, all files will be pushed.
If run from the command palette, only the current file will be pushed.

&nbsp;
The commands & arguments appear to be exactly the same:

```
// @ `Context.sublime-menu`

{ "caption": "Add & Commit & Push…",             "command": "side_bar_git_add_commit_push", "args": {"paths": []} },


// @ `Commands.sublime-commands`

{ "caption": "SideBarGit: Add & Commit & Push…", "command": "side_bar_git_add_commit_push", "args": {"paths": []} },
```

&nbsp;

----------
:question:

&nbsp;

Is there any way to create custom entries @ `Commands.sublime-settings` that will affect the entire repo, rather than only the currently active file?

&nbsp;

:question:

----------

----------

&nbsp;

### Steps To Reproduce:

* Add `test SideBarGit.txt` to repo
* Open a random file ( `CustomCompletions.py` ) from the same repo
* Run `SideBarGit: Add & Commit & Push…`
 * from the command palette
 * OR from the sidebar menu after right-clicking `CustomCompletions.py`

&nbsp;

Those steps result in the following message:

![30f527985f8df062a61bceb2632d4e866024d1a3](https://cloud.githubusercontent.com/assets/10906415/14409316/b014a6ae-fedc-11e5-903a-8f81e57c182a.jpg)

&nbsp;

However, If I run `SideBarGit: Add & Commit & Push…` from the sidebar menu after right-clicking on:
* the repo *directory* `CustomCompletions`
* `test SideBarGit.txt`

I will get the following message:

![a16520663a4de5226b711a1ceb7e0e9766851c13](https://cloud.githubusercontent.com/assets/10906415/14409317/b79d1f0a-fedc-11e5-9db0-2576a46d5d2f.jpg)

&nbsp;

-----

&nbsp;

In short, the new file will only get pushed if you run the command from the sidebar context menu after right clicking on [A] the new file or [B] the parent directory of the new file.

I think that is a great option to have, as it allows very specific control; **but** I also think there should be parameters for the commands that allow them to affect entire repos.
